src/main/java/org/terasology/climateConditions/FrostbiteSystem.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.climateConditions;
+
+import org.terasology.alterationEffects.speed.StunAlterationEffect;
+import org.terasology.audio.StaticSound;
+import org.terasology.audio.events.PlaySoundEvent;
+import org.terasology.context.Context;
+import org.terasology.entitySystem.entity.EntityRef;
+import org.terasology.entitySystem.entity.lifecycleEvents.BeforeRemoveComponent;
+import org.terasology.entitySystem.entity.lifecycleEvents.OnAddedComponent;
+import org.terasology.entitySystem.event.ReceiveEvent;
+import org.terasology.entitySystem.prefab.Prefab;
+import org.terasology.entitySystem.prefab.PrefabManager;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterMode;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.logic.characters.CharacterSoundComponent;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.PeriodicActionTriggeredEvent;
+import org.terasology.logic.health.event.DoDamageEvent;
+import org.terasology.registry.In;
+import org.terasology.utilities.random.FastRandom;
+import org.terasology.utilities.random.Random;
+
+@RegisterSystem(RegisterMode.AUTHORITY)
+public class FrostbiteSystem extends BaseComponentSystem {
+    public static final String FROSTBITE_DAMAGE_ACTION_ID = "Frostbite Damage";
+
+    @In
+    private DelayManager delayManager;
+    @In
+    private PrefabManager prefabManager;
+    @In
+    private Context context;
+
+    private static final int healthDecreaseInterval = 20000;
+    private static final int initialDelay = 5000;
+    private static final int healthDecreaseAmount = 15;
+    private Random random = new FastRandom();
+
+    @ReceiveEvent(components = {HypothermiaComponent.class})
+    public void onHypothermia(OnAddedComponent event, EntityRef player) {
+        delayManager.addPeriodicAction(player, FROSTBITE_DAMAGE_ACTION_ID, initialDelay, healthDecreaseInterval);
+    }
+
+    @ReceiveEvent(components = {HypothermiaComponent.class})
+    public void BeforeRemoveHypothermia(BeforeRemoveComponent event, EntityRef player, HypothermiaComponent hypothermia) {
```

Comment:
This is a method and should start with a lower-case letter.
```suggestion
    public void beforeRemoveHypothermia(BeforeRemoveComponent event, EntityRef player, HypothermiaComponent hypothermia) {
```

Reply:
Also, as you have the component in the annotation and you don't access it in the method body, it should be removed from the parameter list.
```suggestion
    public void beforeRemoveHypothermia(BeforeRemoveComponent event, EntityRef player) {
```